graylog2-web-interface/src/views/components/Search.tsx

```diff
@@ -175,52 +175,56 @@ const Search = ({ location }: Props) => {
-                        <ConnectedSidebar>
-                          <FieldsOverview />
-                        </ConnectedSidebar>
-                      </IfInteractive>
-                      <SearchArea>
-                        <IfInteractive>
-                          <HeaderElements />
-                          <IfDashboard>
-                            <DashboardSearchBarWithStatus onExecute={refreshIfNotUndeclared} />
-                          </IfDashboard>
-                          <IfSearch>
-                            <SearchBarWithStatus onExecute={refreshIfNotUndeclared} />
-                          </IfSearch>
-
-                          <QueryBarElements />
-
-                          <IfDashboard>
-                            <QueryBar />
-                          </IfDashboard>
-                        </IfInteractive>
-                        <HighlightMessageInQuery>
-                          <SearchResult />
-                        </HighlightMessageInQuery>
-                      </SearchArea>
-                    </GridContainer>
-                  </HighlightingRulesProvider>
-                </ViewAdditionalContextProvider>
-              </DefaultFieldTypesProvider>
-            </SearchPageLayoutProvider>
-          )}
-        </InteractiveContext.Consumer>
-      </CurrentViewTypeProvider>
+      <WidgetFocusContext.Consumer>
+        {({ focusedWidget: { focusing: focusingWidget } = { focusing: false } }) => (
+          <CurrentViewTypeProvider>
+            <IfInteractive>
+              <IfDashboard>
+                <WindowLeaveMessage />
+              </IfDashboard>
+            </IfInteractive>
+            <InteractiveContext.Consumer>
+              {(interactive) => (
+                <SearchPageLayoutProvider>
+                  <DefaultFieldTypesProvider>
+                    <ViewAdditionalContextProvider>
+                      <HighlightingRulesProvider>
+                        <GridContainer id="main-row" interactive={interactive}>
+                          <IfInteractive>
+                            <ConnectedSidebar>
+                              <FieldsOverview />
+                            </ConnectedSidebar>
+                          </IfInteractive>
+                          <SearchArea>
+                            <IfInteractive>
+                              <HeaderElements />
+                              <IfDashboard>
+                                <DashboardSearchBarWithStatus onExecute={refreshIfNotUndeclared} />
+                              </IfDashboard>
+                              <IfSearch>
+                                <SearchBarWithStatus onExecute={refreshIfNotUndeclared} />
+                              </IfSearch>
+
+                              <QueryBarElements />
+
+                              <IfDashboard>
+                                {!focusingWidget && <QueryBar />}
+                              </IfDashboard>
```

Comment:
I thought about moving the dashboard and widget focus related logic inside the `QueryBar` component. This would make it easier to test the different paths. But I prefer the current solution, because it is easier to understand (a developer just needs to have a look at the `Search` component to see the possible paths).
Another benefit, the `QueryBar` component only contains code which is directly related to the query bar.